shared/js/advanced_telemetry_helper.js

```diff
@@ -0,0 +1,190 @@
+
+        message = APP_TELEMETRY_LOG_PREFIX;
+        message += '|' + name + '|';
+        message += HISTOGRAM_COUNT;
+        console.info(message);
+      } else {
+        console.warn('A counter telemetry entry requires one parameter');
+      }
+    }
+  };
+
+  // API for a linear histogram entry.  This function creates the histogram
+  // in the system and prepares it for subsequent .log(value) calls against it.
+  // <name>: Unique name of the metric
+  // <min>: the minimum (non-zero) numeric value for this histogram.
+  // <max>: the maximum numeric value for this histogram
+  // <buckets>: the total number of buckets for this histogram.
+  AdvancedTelemetryHelper.prototype.linear = function(name, min, max, buckets) {
+    this.createHistogram(name, HISTOGRAM_LINEAR, min, max, buckets);
+  };
+
+  // API for an exponential histogram entry.  This function creates the
+  // histogram in the system and prepares it subsequent .log(value) calls
+  // against it.
+  // <name>: Unique name of the metric
+  // <min>: the minimum (non-zero) numeric value for this histogram.
+  // <max>: the maximum numeric value for this histogram
+  // <buckets>: the total number of buckets for this histogram.
+  AdvancedTelemetryHelper.prototype.exp = function(name, min, max, buckets) {
+    this.createHistogram(name, HISTOGRAM_EXP, min, max, buckets);
+  };
+
+  AdvancedTelemetryHelper.prototype.createHistogram =
+  function(name, type, min, max, buckets) {
+    if (typeof ATH.TELEMETRY_ENABLED === 'undefined' || ATH.TELEMETRY_ENABLED) {
+      if (typeof this.metricType !== 'undefined') {
+        console.warn('Cannot redefine histogram.  Must create a new object');
+        return;
+      }
+
+      this.metricType = type;
+      if (typeof name !== 'undefined') {
+        var varCount = 2;
+        var message = APP_TELEMETRY_LOG_PREFIX;
+        // Disallow underscores so telemetry system can parse it.
+        this.metricName = name.replace('_', '-');
+        message += '|' + this.metricName;
+        message += '|' + type + '|';
+        if (typeof min !== 'undefined') { varCount++; }
+        if (typeof max !== 'undefined') { varCount++; }
+        if (typeof buckets !== 'undefined') { varCount++; }
+
+        if (varCount != 5) {
+          console.warn('createHistogram must have 5 params.');
+          return;
+        }
+
+        if (min < 1) {
+          min = 1;
+        }
+        message += min + '|' + max + '|' + buckets;
+        console.info(message);
+      } else {
+        console.warn('A telemetry log entry requires a name.');
+      }
+    }
+  };
+
+  // API for logging against a pre-existing histogram.
+  // Pre-requisite is to call .count(..), .linear(...), or .exp(...) first.
+  // <value>: The value to log against the histogram.
+  //    for a counter histogram, this can be empty or 1 can be passed
+  //    for .exp or .linear a valid value > 0 must be passed.
+  AdvancedTelemetryHelper.prototype.log = function(value) {
```

Comment:
I think it's better to encode the purpose for the API function in the name of the function as opposed to how it's being implemented. 

Nit: The comment is also describing the implementation as opposed to the purpose.